core/domain/stats_domain.py

```diff
@@ -63,13 +66,119 @@
 MAX_ANSWER_DETAILS_BYTE_SIZE = 10000
 
 
+class SubmittedAnswerDict(TypedDict, total=False):
+    """Dictionary representing the SubmittedAnswer object."""
+
+    answer: str
+    interaction_id: str
+    answer_group_index: int
+    rule_spec_index: int
```

Comment:
If the answer group is the default one, what is the value of rule_spec_index?

Reply:
I'm not entirely sure, could you please clarify? Thank you!